Framework/PythonInterface/test/python/plugins/algorithms/SaveP2DTest.py

```diff
@@ -0,0 +1,96 @@
+
+        # used to join all spectra
+        xDataTotal = []                                          # d Values for all spectra
+        zDataTotal = []                                          # intensity values for all spectra
+        eDataTotal = []                                          # error values for all spectra
+        nSpec = len(yData)-1                                     # number of spectra
+
+        # Create d and intensity lists for workspace
+        for i in range(0,nSpec):
+            xDataTotal.extend(xData)       # extends the list of x values in accordance to the number of spectra used
+            zDataTotal.extend(zData)       # extends the list of intensity values in accordance to the number of spectra used
+            eDataTotal.extend(eData)       # extends the list of error values in accordance to the number of spectra used
+
+        # Create a 2D Workspace containing d and dPerpendicular values with intensities
+        ws = CreateWorkspace(DataX = xDataTotal, DataY = zDataTotal, DataE = eDataTotal, WorkspaceTitle = 'test', NSpec = nSpec, UnitX = 'dSpacing', VerticalAxisUnit = 'dSpacingPerpendicular', VerticalAxisValues = yData)
+        return ws
+
+    def test_savep2d_raw(self):
+         SaveP2D(Workspace = self._workspace, OutputFile = os.path.join(self._test_dir, "test_savep2d_raw"), RemoveNaN = False, RemoveNegatives = False, CutData = False)
+         text = '#Title: test\n#Inst: .prm\n#Binning: ddperp   0.8888889    1.0000000\n#Bank: 1\n#2theta   lambda   d-value   dp-value   counts\n  81.3046911      1.3029352      1.0000000      1.5000000      1.0000000\n  42.5730378      1.4521280      2.0000000      1.5000000      2.0000000\n  28.5401669      1.4789581      3.0000000      1.5000000      3.0000000\n  21.4420009      1.4882141      4.0000000      1.5000000     -1.0000000\n  17.1666094      1.4924723      5.0000000      1.5000000      0.0000000\n  14.3112545      1.4947782      6.0000000      1.5000000            nan\n  12.2697184      1.4961662      7.0000000      1.5000000      3.0000000\n  10.7376523      1.4970660      8.0000000      1.5000000      1.0000000\n   9.5455787      1.4976825      9.0000000      1.5000000      4.0000000\n 147.7039064      1.9210925      1.0000000      2.5000000      1.0000000\n  74.0366222      2.4082809      2.0000000      2.5000000      2.0000000\n  48.4687709      2.4628222      3.0000000      2.5000000      3.0000000\n  36.1141714      2.4797153      4.0000000      2.5000000     -1.0000000\n  28.8035116      2.4871957      5.0000000      2.5000000      0.0000000\n  23.9632304      2.4911738      6.0000000      2.5000000            nan\n  20.5194188      2.4935442      7.0000000      2.5000000      3.0000000\n  17.9428625      2.4950714      8.0000000      2.5000000      1.0000000\n  15.9421282      2.4961135      9.0000000      2.5000000      4.0000000\n'
+         with open(os.path.join(self._test_dir, "ref_savep2d_raw.txt", 'w')) as of:
```

Comment:
These brackets are misalligned, the 'w' flag is being passed into `os.path.join` rather than `open`
```suggestion
         with open(os.path.join(self._test_dir, "ref_savep2d_raw.txt"), 'w') as of:
```